source/features/render-semantic-types.tsx

```diff
@@ -0,0 +1,191 @@
+
+import features from '../feature-manager.js';
+
+function isSemanticCommitTitleElement(commitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Tests, if the title string
+	// starts with a word,
+	// followed by an anything in parentheses,
+	// followed by an optional exclamation mark,
+	// followed by a colon,
+	// followed by anything.
+	return /^\w*(?:\(.*\))?!?:.*/.test(commitTitle);
+}
+
+const defaultCommitTypes = new Set([
+	'feat',
+	'fix',
+	'chore',
+	'docs',
+	'build',
+	'refactor',
+	'test',
+	'ci',
+	'perf',
+]);
+
+function isSemanticCommitTitleTypeDefault(semanticCommitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)?.pop();
+
+	if (!type) {
+		return false;
+	}
+
+	return defaultCommitTypes.has(type);
+}
+
+function extractSemanticCommitTitleTypeAndScope(semanticCommitTitle: string): [string, string?] {
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)!.pop()!;
+
+	// Gets the word in parentheses from the title.
+	const scope = /\((\w*)\)/.exec(semanticCommitTitle)?.pop();
+
+	return [type, scope];
+}
+
+function removeSemanticCommitTitleAndScopeFromTitleElement(semanticCommitTitleElement: HTMLElement): void {
+	const children = semanticCommitTitleElement.childNodes;
+
+	// Remove all children until the child with a colon is found.
+	// Then remove the colon and everything before it.
+	for (const child of children) {
+		if (child.textContent.includes(':')) {
+			child.textContent = child.textContent.split(':').slice(1).join(':');
+			break;
+		}
+
+		child.remove();
+	}
+}
+
+const defaultCommitTypesMapping = new Map([
+	[
+		'feat',
+		{
+			label: 'Feature',
+			colorCssRules: {
+				'--label-r': 14,
+				'--label-g': 138,
+				'--label-b': 22,
+				'--label-h': 123,
+				'--label-s': 81,
+				'--label-l': 29,
+			},
+		},
+	],
+	[
+		'fix',
+		{
+			label: 'Fix',
+			colorCssRules: {
+				'--label-r': 215,
+				'--label-g': 58,
+				'--label-b': 74,
+				'--label-h': 353,
+				'--label-s': 66,
+				'--label-l': 53,
+			},
+		},
+	],
+	[
+		'chore',
+		{
+			label: 'Chore',
+			colorCssRules: {
+				'--label-r': 170,
+				'--label-g': 170,
+				'--label-b': 170,
+				'--label-h': 0,
+				'--label-s': 0,
+				'--label-l': 255,
+			},
+		},
+	],
+	[
+		'refactor',
+		{
+			label: 'Refactor',
+			colorCssRules: {
+				'--label-r': 255,
+				'--label-g': 150,
+				'--label-b': 0,
+				'--label-h': 36,
+				'--label-s': 100,
+				'--label-l': 50,
+			},
+		},
+	],
+	[
+		'build',
+		{
+			label: 'Build',
+			colorCssRules: {
+				'--label-r': 83,
+				'--label-g': 25,
+				'--label-b': 231,
+				'--label-h': 256,
+				'--label-s': 81,
+				'--label-l': 50,
+			},
+		},
+	],
+]);
+
+function createLabelElement(type: string, scope?: string): JSX.Element {
+	const {label, colorCssRules} = defaultCommitTypesMapping.get(type)!;
```

Comment:
I think you don’t actually need the mapping. Only `feat` is shortened and we could arguably leave it as is. Even “first letter uppercase” might not be necessary since we’re following the style of native GitHub labels—and they’re all lowercase by default. Let’s keep it simple for a first iteration

Reply:
GitHub using lowercase by default for issue labels is a mistake done by GitHub. There is no incentive to use lower case. In my repos I have fixed this mistake:

![image](https://github.com/user-attachments/assets/c1c162ad-7e9a-4791-aa49-ab2b833d99f1)

Everywhere else, GitHub uses uppercase:

![image](https://github.com/user-attachments/assets/8ade5b44-a5b2-4dad-8d36-6fa7385faa8c)
![image](https://github.com/user-attachments/assets/66ae62cd-942f-4957-8347-7e61dc0599ef)
![image](https://github.com/user-attachments/assets/38947957-ce3f-469a-b3b8-610370f6ba35)

Arguably, this is better because in English, the first character is written big. The official primer design language also starts with a big letter in labels, furthermore strengthening the fact that GitHub made a mistake and we should use uppercase first letter:

![image](https://github.com/user-attachments/assets/7ef8c68c-40d7-4d1d-9e85-f32a6e3a7f77)

I have added the mapping for scaling reasons. Yes, maybe there's currently only "feat" that would be mapped to "Feature" (and also "perf" to "Performance"), but we don't know how we will scale in the future, which is why I think the mapping makes sense here. It is already used for two names that need to be written out. 

Additionally, GitHub uses "full" words and not abbreviated ones:

![image](https://github.com/user-attachments/assets/8ade5b44-a5b2-4dad-8d36-6fa7385faa8c)
![image](https://github.com/user-attachments/assets/66ae62cd-942f-4957-8347-7e61dc0599ef)
![image](https://github.com/user-attachments/assets/38947957-ce3f-469a-b3b8-610370f6ba35)

I believe shortening terms often impairing readability, and given that only a couple of words are lengthened by just a couple of characters currently, not adding any meaningful or annoying length, I believe we should keep the current mapping:

![image](https://github.com/user-attachments/assets/f210f9fa-2a39-45e2-b640-e3f2b812660a)